src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/discussions/PostDetailsPage.java

```diff
@@ -32,4 +42,69 @@ public PostDetailsPage open() {
   public boolean isPostDetailsListEmpty() {
     return repliesList.isEmpty();
   }
+
+  public boolean isUpvoteButtonVisible() {
+    wait.forElementVisible(upvoteButton);
+    return upvoteButton.isDisplayed();
+  }
+
+  public boolean isReplyUpvoteButtonVisible(int index) {
+    WebElement button = replyUpvoteButton.get(index);
+    wait.forElementVisible(button);
+    return button.isDisplayed();
+  }
+
+  public void clickPostDetailsUpvoteButton() {
+    wait.forElementClickable(upvoteButton);
+    upvoteButton.click();
+  }
+
+  public void clickReplyUpvoteButton(int replyIndex) {
+    WebElement button = replyUpvoteButton.get(replyIndex);
+    wait.forElementClickable(button);
+    button.click();
+  }
+
+  public String getPostDetailsVoteCount() {
+    wait.forElementVisible(upvoteArea);
+    return upvoteArea.getText();
+  }
+
+  public String getReplyVoteCount(int index) {
+    WebElement voteArea = replyVoteCount.get(index);
+    wait.forElementVisible(voteArea);
+    return voteArea.getText();
+  }
+
+  public void waitForPostDetailsVoteCountToChange(String voteCount) {
+    wait.forTextNotInElement(upvoteArea, voteCount);
+  }
+
+  public void waitForReplyVoteCountToChange(int replyIndex, String voteCount) {
+    WebElement voteArea = replyVoteCount.get(replyIndex);
+    wait.forTextNotInElement(voteArea, voteCount);
+  }
+
+  /**
+   * Wait for the noticeable time lag between vote and vote value change to pass
+   */
+  public void waitForVoteCountChangeTimeLagToPass() {
+    try {
+      //This sleep was introduced because of noticeable lag between vote and vote value change
+      Thread.sleep(2000);
+    } catch (InterruptedException e) {
+      e.printStackTrace();
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Use a logger to log this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1148)